packages/build/src/core/commands.js

```diff
@@ -242,14 +266,35 @@ const fireBuildCommand = async function({ buildCommand, configPath, buildDir, no
     return {}
   } catch (newError) {
     handleBuildCommandOutput(newError, logs)
-    addErrorInfo(newError, { type: 'buildCommand', location: { buildCommand, configPath } })
+    handleBuildCommandError({ error: newError, buildCommand, configPath, env, sitePackageJson, logs })
     return { newError }
   }
 }
 
 // We use Bash on Unix and `cmd.exe` on Windows
 const SHELL = platform === 'win32' ? true : 'bash'
 
+// When `build.command` fails
+const handleBuildCommandError = function({ error, buildCommand, configPath, env, sitePackageJson, logs }) {
+  addErrorInfo(error, { type: 'buildCommand', location: { buildCommand, configPath } })
+
+  if (isCiReactError({ error, env, sitePackageJson })) {
+    logCiReactWarning(logs)
+  }
+}
+
+const isCiReactError = function({ error, env: { CI }, sitePackageJson }) {
+  return (
+    error.exitCode === 1 && typeof CI === 'string' && CI.toLowerCase() !== 'false' && isCreateReactApp(sitePackageJson)
+  )
+}
+
+const isCreateReactApp = function({ scripts }) {
+  return isPlainObj(scripts) && scripts.build === CREATE_REACT_APP_BUILD
```

Comment:
https://github.com/timarney/react-app-rewired is also a common pattern for running `create-react-app` build.
Should we use `scripts.build.startsWith` just in case a user chains commands using `&&` ?

Reply:
Good point! Fixed in 63b138166e6aeb081c01c7645ad370ee40a8f7b2